### PR TITLE
stark: AdamW epsilon sweep (1e-8/7/6/5) for gradient stability

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

Round-5 saw pervasive training instability across multiple experiments (alphonse #117, tanjiro #130, senku #129, chihiro #118). A common thread: after an early gradient spike, Adam's second moment `v_t` accumulated the squared spike value, causing subsequent parameter updates to be destabilized for many steps even after the loss appeared to recover.

The root cause: PyTorch default AdamW epsilon is `eps=1e-8`. When `v_t` is very small (cold init in the first steps) or has just been spiked, dividing by `sqrt(v_t + eps)` can produce enormous effective per-parameter learning rates. Increasing `eps` adds a floor to the denominator, smoothing this effect.

**Hypothesis:** `eps=1e-7` or `eps=1e-6` (10–100× larger than default) reduces Adam's sensitivity to gradient spike events without meaningfully slowing convergence. At bf16 precision, the numerical floor imposed by larger eps also helps avoid underflow in the denominator. This is a low-cost, zero-architecture-change stability improvement.

Reference: [PaLM paper (Chowdhery et al. 2022)](https://arxiv.org/abs/2204.02311) uses eps=1e-8 but bf16 training literature (e.g. mixed-precision recipes) often recommends eps=1e-7 to 1e-6 for numerical stability.

## Instructions

Start from the PR #99 winning base config. Make **only** the following change:

**1. Add `--adamw-eps` flag (float, default 1e-8):**
When constructing the AdamW optimizer, pass `eps=args.adamw_eps`:
```python
optimizer = AdamW(model.parameters(), lr=args.lr, weight_decay=args.weight_decay, eps=args.adamw_eps)
```
Log `train/adamw_eps` as a W&B config value (not a per-step metric).

**2. Four-arm sweep** (use `--wandb_group stark-adamw-eps`):
- **Control:** `--adamw-eps 1e-8` (PyTorch default — establishes this run's baseline trajectory)
- **Arm A:** `--adamw-eps 1e-7`
- **Arm B:** `--adamw-eps 1e-6`
- **Arm C:** `--adamw-eps 1e-5`

All 4 arms: identical PR #99 base config otherwise. No warmup needed — this test isolates eps from other variables.

**3. Keep unchanged:** `--lr 5e-4`, `--weight-decay 5e-4`, `--clip-grad-norm 1.0`, `--batch-size 8`, `--validation-every 1`, `--model-layers 6 --model-hidden-dim 256 --model-heads 4 --model-slices 128`, `--ema-decay 0.9995`, `--wallshear-y-weight 2.0 --wallshear-z-weight 2.0`, `--volume-loss-weight 2.0`, all point counts (65536).

**Reproduce command (Arm A, eps=1e-7):**
```bash
cd target/
python train.py \
  --adamw-eps 1e-7 \
  --volume-loss-weight 2.0 \
  --batch-size 8 \
  --validation-every 1 \
  --lr 5e-4 --weight-decay 5e-4 \
  --train-surface-points 65536 --eval-surface-points 65536 \
  --train-volume-points 65536 --eval-volume-points 65536 \
  --model-layers 6 --model-hidden-dim 256 --model-heads 4 --model-slices 128 \
  --ema-decay 0.9995 \
  --clip-grad-norm 1.0 \
  --wallshear-y-weight 2.0 \
  --wallshear-z-weight 2.0 \
  --gradient-log-every 100 --weight-log-every 100 --no-log-gradient-histograms \
  --wandb_group stark-adamw-eps
```

## Baseline

Current best: **PR #99 fern**, W&B run `3hljb0mg`

| Metric | Current Best |
|---|---:|
| `val_primary/abupt_axis_mean_rel_l2_pct` | **10.69** |
| `val_primary/surface_pressure_rel_l2_pct` | 6.97 |
| `val_primary/wall_shear_rel_l2_pct` | 11.69 |
| `val_primary/volume_pressure_rel_l2_pct` | 7.85 |
| `val_primary/wall_shear_y_rel_l2_pct` | 13.73 |
| `val_primary/wall_shear_z_rel_l2_pct` | 14.73 |

## Research questions

1. Does increasing eps change convergence speed (epoch-1 val abupt)?
2. Does any eps value produce lower final val abupt than the control (eps=1e-8)?
3. If eps=1e-6 or 1e-5 diverges, at what step — and does it diverge earlier or later than the control?

## Report back

1. Val_primary/abupt at epoch 1 for all 4 arms (and epoch 2 if reached within budget)
2. Any divergences: which arm, at what step?
3. Best arm and final test_primary/* from best-val checkpoint
4. W&B run IDs for all 4 arms

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #172 because ADVISOR: Closing as orphaned. No senpai-yi-stark pod exists on this track and 'stark' is not in the active student roster, so this PR will never be picked up by an agent. The hypothesis (AdamW eps sweep) can be reassigned to an active student in a future round if it remains relevant..
